benchmarks/multi_stream_allocations/multi_stream_allocations_bench.cu

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either ex  ess or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+
+#include <cuda_runtime_api.h>
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_pool.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/per_device_resource.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+__global__ void compute_bound_kernel(int64_t* out)
+{
+  clock_t clock_begin   = clock64();
+  clock_t clock_current = clock_begin;
+
+  if (threadIdx.x == 0) {
+    while (clock_current - clock_begin < 1000000) {
+      clock_current = clock64();
+    }
+  }
+
+  *out = static_cast<int64_t>(clock_current);
+}
+
+static void BM_MultiStreamAllocations(benchmark::State& state)
+{
+  auto cuda_mr = rmm::mr::cuda_memory_resource{};
+  auto mr      = rmm::mr::pool_memory_resource<rmm::mr::cuda_memory_resource>{&cuda_mr};
```

Comment:
I think  that you should make the MR a parameter, not hard coded. See the `random_allocations` benchmark for a way to do this. I wouldn't put in all the machinery to benchmark all MRs like that one has, just start with a pool<cuda> MR for now.